assistants/relay-e2e/src/header/eth_beacon_header_relay.rs

```diff
@@ -0,0 +1,233 @@
+use std::{
+    ops::Div,
+    str::FromStr,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use bridge_e2e_traits::client::EthTruthLayerLightClient;
+use client_beacon::{client::BeaconApiClient, types::FinalityUpdate};
+use client_contracts::beacon_light_client_types::FinalizedHeaderUpdate;
+use support_etherscan::wait_for_transaction_confirmation;
+use web3::{
+    contract::Options,
+    types::{Bytes, H256, U256},
+};
+
+use crate::error::{RelayError, RelayResult};
+
+pub struct BeaconHeaderRelayRunner<C: EthTruthLayerLightClient> {
+    pub eth_light_client: C,
+    pub beacon_api_client: BeaconApiClient,
+    pub minimal_interval: u64,
+    pub last_relay_time: u64,
+}
+
+#[derive(Debug)]
+pub struct HeaderRelayState {
+    // Latest relayed header slot at Darwinia
+    pub relayed_slot: u64,
+    // Latest relayed period at Darwinia
+    pub relayed_period: u64,
+    // Latest header slot at Beacon chain
+    pub current_slot: u64,
+    // Latest period at Beacon chain
+    pub current_period: u64,
+}
+
+impl<C: EthTruthLayerLightClient> BeaconHeaderRelayRunner<C> {
+    pub async fn start(&mut self) -> RelayResult<()> {
+        loop {
+            self.run().await?;
+            tokio::time::sleep(std::time::Duration::from_secs(10)).await;
+        }
+    }
+
+    pub async fn run(&mut self) -> RelayResult<()> {
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map_err(|e| RelayError::Custom(format!("{}", e)))?
+            .as_secs();
+        if now - self.last_relay_time <= self.minimal_interval {
```

Comment:
A simple way is use `config.general.header_relay_minimum_interval` in tokio::time::sleep(std::time::Duration::from_secs(10)).await;

tokio::time::sleep(std::time::Duration::from_secs(config.general.header_relay_minimum_interval)).await;

you don't need to write this if condition.